File: synapse-transport/src/main/java/com/synapse/transport/feature/experiment/dto/response/ExperimentModelResponse.java
package com.synapse.transport.feature.experiment.dto.response;

import java.util.UUID;

import tools.jackson.databind.JsonNode;

import com.synapse.transport.common.enums.ExperimentTaskType;
import com.synapse.transport.common.enums.LossFunctionType;
import com.synapse.transport.common.enums.ModelStatus;
import com.synapse.transport.common.enums.OptimizerType;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ExperimentModelResponse {
    private UUID experimentId;
    private ExperimentTaskType taskType;

    private ModelStatus modelStatus;
    private JsonNode modelIr;

    private String inputShape;
    private String outputShape;

    private OptimizerType optimizer;
    private LossFunctionType lossFunction;
    private Double learningRate;
    private Integer epochs;

    private String lastTrainingError;
}

File: synapse-ui/src/types/api/experiment/experiment-model.response.ts
import type {
  ExperimentTaskType,
  LossFunctionType,
  ModelStatus,
  OptimizerType,
} from "../../enums/transport.enums";
import type { ModelIR } from "../../runtime/model/model-ir.type";

export interface ExperimentModelResponse {
  experimentId: string;
  taskType: ExperimentTaskType;
  modelStatus: ModelStatus;
  modelIr: ModelIR | null;
  inputShape: string | null;
  outputShape: string | null;
  optimizer: OptimizerType | null;
  lossFunction: LossFunctionType | null;
  learningRate: number | null;
  epochs: number | null;
  lastTrainingError: string | null;
}
